addon/ng2/utilities/fs-promise.ts

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+import * as fs from 'fs';
+
+/**
+ * Read a file and return a promise
+ * @param path (path to file)
+ * @return Promise with file content
+ */
+export function readWithPromise(path: string): Promise<any> {
+    var content: string;
+    return new Promise<any>((resolve, reject) => {
+      fs.readFile(path, (err, data) => {
+        if(err) reject(err);
```

Comment:
Always use braces:

```
if (err) {
  reject(err);
}
```

You should also `return` after rejecting, otherwise the code as written will try to resolve the promise as well, which is incorrect behavior.